### PR TITLE
nrop pkg: NUMAResourcesOperator handler added

### PR DESCRIPTION
**Pull request by elenagerman**
nrop pkg: NUMAResourcesOperator handler added

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -218,6 +218,8 @@ require (
 	sigs.k8s.io/yaml v1.4.0 // indirect
 )
 
+require github.com/openshift-kni/numaresources-operator v0.1.1
```
Comment (cdvultur) [inline go.mod:221]: from what I see, the v0.1.1 is 3 years old. 

Don't you want to use the one tied with OCP Version ?

**Review by cdvultur — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (elenagerman) [reply go.mod]: yep, good idea, thanks!

**Review by cdvultur — APPROVED** [review]
lgtm

File: pkg/nrop/nrop.go [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+
+		return false, fmt.Errorf(msg.UndefinedCrdObjectErrString(resourceCRD))
+	}
+
+	if builder.apiClient == nil {
+		glog.V(100).Infof("The %s builder apiclient is nil", resourceCRD)
+
+		return false, fmt.Errorf("%s builder cannot have nil apiClient", resourceCRD)
+	}
+
```
Comment (kononovn) [inline pkg/nrop/nrop.go:267]: Please add 1 more validation:

```
	if builder.errorMsg != "" {
		glog.V(100).Infof("The %s builder has error message: %s", resourceCRD, builder.errorMsg)

		return false, fmt.Errorf(builder.errorMsg)
	}
```

**Review by kononovn — CHANGES_REQUESTED** [review]
Please fix 1 comment

**Review by kononovn — APPROVED** [review]
/lgtm

**Pull request merged**